src/tools/clippy/clippy_lints/src/utils/mod.rs

```diff
@@ -1230,6 +1230,27 @@ pub fn match_def_path<'tcx>(cx: &LateContext<'tcx>, did: DefId, syms: &[&str]) -
     cx.match_def_path(did, &syms)
 }
 
+pub fn match_panic_call<'tcx>(
+    cx: &LateContext<'tcx>,
+    expr: &'tcx Expr<'_>,
+) -> Option<&'tcx [Expr<'tcx>]> {
+    match_function_call(cx, begin_panic_call, &paths::BEGIN_PANIC)
+        .or_else(|| match_function_call(cx, begin_panic_call, &paths::BEGIN_PANIC_FMT))
+        .or_else(|| match_function_call(cx, begin_panic_call, &paths::PANIC_ANY))
+        .or_else(|| match_function_call(cx, begin_panic_call, &paths::PANICKING_PANIC))
+        .or_else(|| match_function_call(cx, begin_panic_call, &paths::PANICKING_PANIC_FMT))
+        .or_else(|| match_function_call(cx, begin_panic_call, &paths::PANICKING_PANIC_STR))
+}
+
+pub fn match_panic_def_path<'tcx>(cx: &LateContext<'tcx>, did: DefId) -> bool {
```

Comment:
NIT:

```suggestion
pub fn match_panic_def_id<'tcx>(cx: &LateContext<'tcx>, did: DefId) -> bool {
```

would be the better name here IMO